src/trusted/service_runtime/nacl_signal_common.c

```diff
@@ -182,6 +182,30 @@ void NaClSignalTestCrashOnStartup(void) {
   }
 }
 
+static void NaClUserRegisterStateFromSignalContextTrusted(
+    volatile NaClUserRegisterState *dest,
+    const struct NaClAppThread *src) {
+#define COPY_REG_NATP(reg) dest->reg = src->user.reg
+  dest->rax = 0;
+  dest->rcx = 0;
+  dest->rdx = 0;
+  dest->rdi = 0;
+  dest->rsi = 0;
+  dest->r8 = 0;
+  dest->r9 = 0;
+  dest->r10 = 0;
+  dest->r11 = 0;
+  COPY_REG_NATP(rbx);
+  dest->stack_ptr = src->user.rsp;
+  COPY_REG_NATP(rbp);
+  COPY_REG_NATP(r12);
+  COPY_REG_NATP(r13);
+  dest->r14 = 0;
+  COPY_REG_NATP(r15);
+  dest->flags = 0;
```

Comment:
so why we want drop most registers, how are the registers copied special?

Reply:
sorry I dont understand the question